xonsh/history/base.py

```diff
@@ -73,6 +73,7 @@ def __init__(self, sessionid=None, **kwargs):
         self.hist_size = None
         self.hist_units = None
         self.remember_history = True
+        self.should_append = lambda cmd: True
```

Comment:
this can be a regular method. It will be more explicit and easier to extend if needed.

Reply:
I also suggest naming the method something like, `is_ignored(cmd) -> bool`